### ISSUE TITLE
在自己的训练集上训练，测试集合损失函数降到0.0068了，还是预测不出来结果

### ISSUE THREAD

**Issue by DeeeeepL** [YunYang1994/tensorflow-yolov3#203]
在自己的训练集上训练，测试集合损失函数降到0.0068了，还是预测不出来结果。
=> predict result of 5013.bmp:
=> ground truth of 3022.bmp:
	weld 72 75 120 111
	weld 211 50 270 90
	weld 126 405 215 447
=> predict result of 3022.bmp:
=> ground truth of 744.bmp:
	weld 99 72 137 102
	weld 229 58 285 100
	weld 117 424 197 462
=> predict result of 744.bmp:
=> ground truth of 2512.bmp:
	weld 51 70 102 103
	weld 201 65 254 106
	weld 113 383 193 425
=> predict result of 2512.bmp:
全部图片都预测不出来结果，有遇到过类似问题的吗

**Comment by shoutOutYangJie** [comment]
你解决了吗？我还在训练，但是在test loss中出现了nan

**Comment by shoutOutYangJie** [comment]
In evaluate.py file， please set "trainable" as True.

**Comment by DeeeeepL** [comment]
> In evaluate.py file， please set "trainable" as True.

self.trainable: True请问这样设置的目的是什么呀

**Comment by shoutOutYangJie** [comment]
because batchnorm use the flag : trainable, which means that there is a difference bewteen training process and test phase


 
---原始邮件---
发件人: "DeeeeepL"<notifications@github.com>
发送时间: 2019年6月26日 13:20:43
收件人: "YunYang1994/tensorflow-yolov3"<tensorflow-yolov3@noreply.github.com>;
抄送: "Comment"<comment@noreply.github.com>;"shoutOutYangJie"<312358434@qq.com>;
主题: Re: [YunYang1994/tensorflow-yolov3] 在自己的训练集上训练，测试集合损失函数降到0.0068了，还是预测不出来结果 (#203)


 
In evaluate.py file， please set "trainable" as True.
  
self.trainable: True请问这样设置的目的是什么呀

**Comment by freemanls** [comment]
个人建议：查询一下训练集是否有问题，在转换为annocation转换txt文档的时候，用图片程序跑下里面的数据是否能支持画框的正确。

**Comment by DeeeeepL** [comment]
> because batchnorm use the flag : trainable, which means that there is a difference bewteen training process and test phase
> […](#)
> ---原始邮件--- 发件人: "DeeeeepL"<notifications@github.com> 发送时间: 2019年6月26日 13:20:43 收件人: "YunYang1994/tensorflow-yolov3"<tensorflow-yolov3@noreply.github.com>; 抄送: "Comment"<comment@noreply.github.com>;"shoutOutYangJie"<312358434@qq.com>; 主题: Re: [YunYang1994/tensorflow-yolov3] 在自己的训练集上训练，测试集合损失函数降到0.0068了，还是预测不出来结果 (#203) In evaluate.py file， please set "trainable" as True. self.trainable: True请问这样设置的目的是什么呀 — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or mute the thread.

Thank you for your suggestion, I have solved it.

**Comment by rw1995** [comment]
> > because batchnorm use the flag : trainable, which means that there is a difference bewteen training process and test phase
> > […](#)
> > ---原始邮件--- 发件人: "DeeeeepL"[notifications@github.com](mailto:notifications@github.com) 发送时间: 2019年6月26日 13:20:43 收件人: "YunYang1994/tensorflow-yolov3"[tensorflow-yolov3@noreply.github.com](mailto:tensorflow-yolov3@noreply.github.com); 抄送: "Comment"[comment@noreply.github.com](mailto:comment@noreply.github.com);"shoutOutYangJie"[312358434@qq.com](mailto:312358434@qq.com); 主题: Re: [YunYang1994/tensorflow-yolov3] 在自己的训练集上训练，测试集合损失函数降到0.0068了，还是预测不出来结果 (#203) In evaluate.py file， please set "trainable" as True. self.trainable: True请问这样设置的目的是什么呀 — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or mute the thread.
> 
> Thank you for your suggestion, I have solved it.

很好奇是什么原因，请问方便说一下吗？

**Comment by tensorflowt** [comment]
你好！能问一下你的first_stage_train_epochs、second_stage_train_epochs分别设定为多少？

**Comment by wuranran** [comment]
> ur suggestion, I have solved it.

请问您如何解决

**Comment by WantYKnow** [comment]
难道就是把Trainable 设置为TRUE吗？

**Comment by navy63** [comment]
好招人恨，说话说一半，

**Comment by navy63** [comment]
> 难道就是把Trainable 设置为TRUE吗？
是的，设置True就可以，亲测过，

**Comment by jiarenyf** [comment]
> > 难道就是把Trainable 设置为TRUE吗？
> > 是的，设置True就可以，亲测过，

怎么能把evaluate中bn层的trainabel改为true呢。。。

**Comment by wl4135** [comment]
我觉得可以多训练一些epochs就可以了，单单改trainable为true的话转换为pb文件会不会检测效果很差呢？
